src/main/java/org/spongepowered/api/map/MapView.java

```diff
@@ -0,0 +1,226 @@
+     */
+    boolean isDoingAutomaticUpdates();
+
+    /**
+     * Sets if vanilla behavior for map updates (slowing updating a few rows
+     * at a time) is being used. The only way a map will update if set to false
+     * is if plugins call on of the {@link #sendUpdate} methods manually, which
+     * could be useful for maps that should constantly update.
+     *
+     * @param doesAutomaticUpdates True if vanilla behavior should be used,
+     *        false otherwise
+     */
+    void setDoesAutomaticUpdates(boolean doesAutomaticUpdates);
+
+    /**
+     * Returns if before using any {@link MapRenderer}s the {@link MapCanvas} should
+     * have the vanilla map copied into it so plugins can do overlay rendering on maps
+     * without implementing vanilla map behavior themselves.
+     *
+     * @return True if vanilla rendering is used as a base, false otherwise
+     */
+    boolean isUsingVanillaRendering();
+
+    /**
+     * Sets if vanilla's map rendering should be added to the {@link MapCanvas} before
+     * plugin {@link MapRenderer}s are run to allow overlaying.
+     *
+     * @param usesVanillaRendering True to enable vanilla rendering as a base, false
+     *        otherwise
+     */
+    void setUsingVanillaRendering(boolean usesVanillaRendering);
+
+    /**
+     * Adds a renderer to this map as the top layer.
+     *
+     * @param renderer The renderer to add
+     */
+    void addRenderer(MapRenderer renderer);
+
+    /**
+     * Inserts a renderer in the map at the specific index, shifting existing
+     * renderers around it.
+     *
+     * @param index The index to insert the renderer at
+     * @param renderer The renderer to insert
+     */
+    void insertRenderer(int index, MapRenderer renderer);
+
+    /**
+     * Removes the specified renderer from the rendering list.
+     *
+     * @param renderer The renderer to remove
+     * @return True if the renderer was successful to be removed, false
+     *         otherwise
+     */
+    boolean removeRenderer(MapRenderer renderer);
+
+    /**
+     * Removes the {@link MapRenderer} at the specified index from the
+     * renderers on this map
+     *
+     * @param index The index of the renderer to remove
+     * @return True if the removal was successful, false otherwise
+     */
+    boolean removeRenderer(int index);
+
+    /**
+     * Removes all renderers from this map.
+     */
+    void clearRenderers();
+
+    /**
+     * Returns a {@link List} of all the {@link MapRenderer}s stored on this
+     * map. The ordering is the layering of rendering with 0 being the bottom
+     * most layer rendered first, and the {@link List#size()}-1 entry being the
+     * latest rendered layer.
+     *
+     * @return An unmodifiable ordered list of all renderers
+     */
+    ImmutableList<MapRenderer> getAllRenderers();
+
+    /**
+     * Returns an {@link ImmutableList} of all the {@link Player}s viewing this
+     * map currently.
```

Comment:
This method is actually not implementable, at least not conveniently, removing it in next commit. There are too many ways to consider a map in view, and the client reports none of them.